File: w09/d01/joshua_oynick/mail_items/spec/mailitem_spec.rb
require_relative 'spec_helper'
require_relative '../lib/mailitem'

describe Mailitem do

  it "has a ship_to_address" do
    mail_item = Mailitem.new(ship_to_address: '1 North Pole')
    expect(mail_item.ship_to_address).to eq('1 North Pole')
  end

  it "has a stamp value" do
    mail_item = Mailitem.new(stamp_value: 'yomomma')
    expect(mail_item.stamp_value).to eq 'yomomma'
  end

end

File: w09/d03/quin_cogdell/mail_objects_app/spec/javascripts/mailitem_spec.js
describe("a mailItem",function(){
  var mail, address, stampValue;

  beforeEach(function(){
    mail = new mailItem();
    address = "123 North Pole";
    addressBad = 123;
    stampPrice = 0.40;
    stampPriceBad = "0.40";
  });

  it("has a shipToAddress", function(){
    mail.shipToAddress(address);
    expect(mail.shipToAddress).toBe("123 North Pole");
  });

  it("has a shipToAddress that is a string", function(){
    var testAddress = function() {
      mail.shipToAddress(addressBad);
    };
    expect(testAddress).toThrow();
  });

  it("has a stampValue", function(){
    mail.stampValue(stampPrice);
    expect(mail.stampValue).toBe(0.40);
  });

  it("has a shipToAddress that is a string", function(){
    var testStampValue = function() {
      mail.stampValue(stampPriceBad);
    };
    expect(testStampValue).toThrow();
  });


});



